### PR TITLE
chore: cleanup jumpstart factory

### PR DESCRIPTION
**Pull request by evakravi**
*Issue #, if available:*

*Description of changes:*
The factory modules for JumpStart contain a lot of duplicate code, which introduces room for error and reduces code maintainability. In addition, there were places where the model type wasn't being piped to the lower level utilities, which is a result of having messy code. This PR reduces the duplication of the code and also fixes places where the model type wasn't being used. This PR will make it easier in the future to add fields such as `config_name`, `model_type` and not have to update as many places in the code. 

*Testing done:*
Current unit + integ tests pass. 

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [ ] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [ ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/jumpstart/artifacts/hyperparameters.py [hunk 1]
```diff
@@ -38,6 +39,7 @@ def _retrieve_default_hyperparameters(
     sagemaker_session: Session = DEFAULT_JUMPSTART_SAGEMAKER_SESSION,
     instance_type: Optional[str] = None,
     config_name: Optional[str] = None,
+    model_type: JumpStartModelType = JumpStartModelType.OPEN_WEIGHTS,
```
Comment (Captainia) [inline src/sagemaker/jumpstart/artifacts/hyperparameters.py:42]: Thanks for back-filling these missing args!

File: src/sagemaker/jumpstart/factory/utils.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+KwargsType = Union[
+    JumpStartModelDeployKwargs,
+    JumpStartModelInitKwargs,
+    JumpStartEstimatorFitKwargs,
+    JumpStartEstimatorInitKwargs,
+    JumpStartEstimatorDeployKwargs,
+]
+
+
+def get_model_info_kwargs(
```
Comment (Captainia) [inline src/sagemaker/jumpstart/factory/utils.py:36]: How about `get_model_info_default_kwargs`, potentially there are other kwargs but these are passed by default to each calls

File: src/sagemaker/jumpstart/factory/estimator.py [hunk 3]
```diff
@@ -263,6 +283,22 @@ def get_fit_kwargs(
         config_name=config_name,
     )
 
+    estimator_fit_kwargs = _set_temp_sagemaker_session_if_not_set(kwargs=estimator_fit_kwargs)
+    estimator_fit_kwargs.specs = verify_model_region_and_return_specs(
```
Comment (Captainia) [inline src/sagemaker/jumpstart/factory/estimator.py:287]: Can we use `get_model_info_kwargs` here as well?

**Review by Captainia — APPROVED** [review]

**Review by akrishna1995 — APPROVED** [review]

File: src/sagemaker/jumpstart/factory/utils.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+def get_model_info_default_kwargs(
+    kwargs: KwargsType,
+    include_config_name: bool = True,
+    include_model_version: bool = True,
+    include_tolerate_flags: bool = True,
+) -> dict:
+    """Returns a dictionary of model info kwargs to use with JumpStart APIs."""
+
+    kwargs_dict = {
```
Comment (JGuinegagne) [inline src/sagemaker/jumpstart/factory/utils.py:45]: future PRs: consider defining type `KwargsType` here

**Pull request merged**